### DISCUSSION TITLE
Usage of Generated .fmu file

### DISCUSSION THREAD

**Discussion by StridingToyota** [NTNU-IHB/PythonFMU#141] [Q&A]
Hello!

I am trying to use my generated .fmu file in an external software. In my case the it's Amesim, however I get errors with others like Simpack or Simulink. 

As a test, I generated the Resistor.py as an .fmu following step by step the example description. The file is generated just fine, however, when it comes to its use it seems to fail with the before mentioned softwares. They all give a different error so I am not going to paste it unless someone deems it necessary. 

Anyone has any advice on how to use these created fmu's in other platforms?

Thank you for the awesome tool and taking your time to read this :)

G.

**Comment by markaren** [comment]
Windows or Linux? In any case, Python needs to be on PATH.

Linux has its issues. More specifically that you need to replace the .so file in the binary if you are not running Python 3.9.

Another prominent issue is loading PythonFMU FMUs in tools that already has loaded the Python shared library (because the tool itself uses Python) and that .dll version is incompatible with the one expected by PythonFMU.

If you are on Windows, try loading the FMU in fmpy.gui: `python -m fmpy.gui` 

Someone else with Amesim issues: https://github.com/NTNU-IHB/PythonFMU/issues/112 (no apparent solution there)
